L1Trigger/L1THGCal/test/calib/testCalibration.cc

```diff
@@ -0,0 +1,159 @@
+#include "DataFormats/Common/interface/Handle.h"
+#include "DataFormats/L1THGCal/interface/HGCalTriggerCell.h"
+#include "CommonTools/UtilAlgos/interface/TFileService.h"
+#include "L1Trigger/L1THGCal/interface/HGCalTriggerBackendProcessor.h"
+#include "TTree.h"
+
+//
+// class declaration
+//
+using namespace std;
+using namespace edm;
+using namespace reco;
+using namespace HGCalTriggerBackend;
+using namespace l1t;
+
+class testCalibration : public edm::EDAnalyzer {
+
+public:
+ 
+    explicit testCalibration(const edm::ParameterSet& );
+    
+    ~testCalibration();
+  
+private:
+ 
+    edm::Service<TFileService> fs_;  
+    virtual void analyze(const edm::Event&, const edm::EventSetup&);
+  
+    // ----------member data ---------------------------
+    EDGetTokenT<HGCalTriggerCellBxCollection > tokenTrgCell_;
+   
+    bool debug_;
+    int _nEvent=0;
+
+    TH1D *h_photon_pt_;
+    TH1D *h_tc_hwPt_;
+    TH1D *h_tc_E_;
+    TH1D *h_tc_pt_;
+    TH1D *h_tc_eta_;
+    TH1D *h_tc_phi_;
+    TH1D *h_tc_layer_;
+
+    TTree *mytree_;  
+
+    double _TC_pt=0.;
+};
+
+
+
+// constructors and destructor
+testCalibration::testCalibration(const edm::ParameterSet& iConfig) : 
+    tokenTrgCell_( consumes< HGCalTriggerCellBxCollection >( iConfig.getParameter<InputTag> ( "triggerCellInputTag" ) ) ),
+    debug_( iConfig.getParameter<bool>( "DebugCode" ) )
+{
+    
+//    consumes< BXVector< l1t::HGCalTriggerCell >  > (TrgCells_tag_);    
+//    consumes< HGCalTriggerCellBxCollection > (TrgCells_tag_);    
+   
+    //now do what ever initialization is needed
+    //edm::Service<TFileService> fs;
+    h_photon_pt_ = fs_->make<TH1D>( "h_photon_pt" ,  "h_photon_pt" , 240 , 0 , 120 );
+    h_tc_hwPt_   = fs_->make<TH1D>("h_tc_hwPt" , "h_tc_hwPt" , 100 , 0 , 2 );
+    h_tc_E_      = fs_->make<TH1D>("h_tc_E" , "h_tc_E" , 200 , 0 , 50 );
+    h_tc_pt_     = fs_->make<TH1D>("h_tc_pt" , "h_tc_pt" , 200 , 0 , 50 );
+    h_tc_eta_    = fs_->make<TH1D>("h_tc_eta" , "h_tc_eta" , 100 , -3.1 , 3.1 );
+    h_tc_phi_    = fs_->make<TH1D>("h_tc_phi" , "h_tc_phi" , 100 , -3.14 , 3.14 );
+    h_tc_layer_  = fs_->make<TH1D>("h_tc_layer" , "h_tc_layer" , 40 , 0 , 40 );
+    
+    mytree_     = fs_->make <TTree>("tree","tree"); 
+
+    //TTree branch
+    mytree_->Branch("event",	&_nEvent,	"event/I");
+    mytree_->Branch("TC_pt",	&_TC_pt,	"TC_pt/D");
+
+}
+
+
+testCalibration::~testCalibration()
+{
+}
+
+
+// ------------ method called for each event  ------------
+void testCalibration::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup)
+{
+    using namespace std;
+    using namespace edm;
+    using namespace reco;
+    using namespace HGCalTriggerBackend;
+    using namespace l1t;
+
+    Handle<HGCalTriggerCellBxCollection>  trgCell;
+    iEvent.getByToken(tokenTrgCell_,  trgCell);
+        
+//    Handle< >;
+//    iEvent.getByLabel(TrgCells_tag_, trgCell);
+   
+    double E_allTC_endcap0=0.;
+    double E_allTC_endcap1=0.;
```

Comment:
Use clearer names instead of `endcap0` and `endcap1`.  For instance, `positive` and `negative`